WordPress/Classes/Services/NotificationsService.swift

```diff
@@ -0,0 +1,223 @@
+    *  @param       wordPressComApi         The WordPressComApi that should be used.
+    */
+    public convenience init(managedObjectContext context: NSManagedObjectContext, wordPressComApi: WordPressComApi) {
+        self.init(managedObjectContext: context)
+        self.remoteApi = wordPressComApi
+    }
+    
+    
+    /**
+    *  @details     This method will retrieve all of the Notification Settings for the default WordPress.com account
+    *  @param       success Closure to be called on success.
+    *  @param       failure Closure to be called on failure, with the associated error.
+    */
+    public func getAllSettings(success: ([NotificationSettings] -> Void)?, failure: (NSError! -> Void)?) {
+        notificationsServiceRemote?.getAllSettings(deviceId,
+            success: {
+                (remote: [RemoteNotificationSettings]) in
+                let parsed = self.settingsFromRemote(remote)
+                success?(parsed)
+            },
+            failure: { (error: NSError!) in
+                failure?(error)
+            })
+    }
+    
+    
+    /**
+    *  @details     Updates the specified NotificationSettings's Stream, with a collection of new values.
+    *  @param       settings    The NotificationSettings to be updated.
+    *  @param       stream      Reference to the specific Settings Stream to update.
+    *  @param       newValues   The collection of new values to be submited.
+    *  @param       success     Closure to be called on success.
+    *  @param       failure     Closure to be called on failure, with the associated error.
+    */
+    public func updateSettings(settings: NotificationSettings, stream: Stream, newValues: [String: Bool], success: (() -> ())?, failure: (NSError! -> Void)?) {
+        let remote = remoteFromSettings(newValues, channel: settings.channel, stream: stream)
+        let pristine = stream.preferences
+        
+        // Preemptively Update the new settings
+        for (key, value) in newValues {
+            stream.preferences?[key] = value
+        }
+        
+        notificationsServiceRemote?.updateSettings(remote,
+            success: {
+                success?()
+            },
+            failure: { (error: NSError!) in
+                // Fall back to Pristine Settings
+                stream.preferences = pristine
+                failure?(error)
+            })
+    }
+
+    
+    /**
+    *  @details     Static Helper that will parse RemoteNotificationSettings instances into a collection of
+    *               NotificationSettings instances.
+    *  @param       remoteSettings  Array of RemoteNotificationSettings
+    *  @returns                     An array of NotificationSettings objects
+    */
+    private func settingsFromRemote(remoteSettings: [RemoteNotificationSettings]) -> [NotificationSettings] {
+        var parsed       = [NotificationSettings]()
+        let blogMap      = blogService.blogsForAllAccountsById() as? [Int: Blog]
```

Comment:
It makes sense, since -now that you mention it- the NotificationSettings, per se, come from the WordPress.com backend, which is expected to be tied up to the main account.

Thanks for checking this one @koke, i'll review this one in another PR.